atomicapp/requirements.py

```diff
@@ -79,12 +79,12 @@ def _exec(self, action):
             requirement_function = self._find_requirement_function_name(key_name)
 
             # Check to see if the function exists in the provider,
-            # if it does not: fail
+            # if it does not: warn the user
             try:
                 requirement = getattr(self.provider, requirement_function)
             except AttributeError:
-                raise RequirementFailedException(
-                    "Requirement %s does not exist within %s." %
+                return logger.warning(
+                    "Requirement %s does not exist within %s. Skipping." %
                     (requirement_function, self.provider))
```

Comment:
and actually.. now that I look at the code closer I don't think we should return at all, but rather continue on to the next iteration of the loop. I missed that it was a loop before.